x-pack/plugins/file_upload/public/components/geojson_upload_form/index_name_form.tsx

```diff
@@ -5,23 +5,46 @@
  * 2.0.
  */
 
+import _ from 'lodash';
 import React, { ChangeEvent, Component } from 'react';
 import { EuiFormRow, EuiFieldText, EuiCallOut, EuiSpacer } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
-import { validateIndexName } from '../../util/indexing_service';
+import { validateIndexName } from '../../validate_index_name';
 
 export interface Props {
   indexName: string;
   indexNameError?: string;
   onIndexNameChange: (name: string, error?: string) => void;
+  onIndexNameValidationStart: () => void;
+  onIndexNameValidationEnd: () => void;
 }
 
 export class IndexNameForm extends Component<Props> {
-  _onIndexNameChange = async (event: ChangeEvent<HTMLInputElement>) => {
+  private _isMounted = false;
+
+  componentDidMount() {
+    this._isMounted = true;
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  _onIndexNameChange = (event: ChangeEvent<HTMLInputElement>) => {
     const indexName = event.target.value;
+    this.props.onIndexNameChange(indexName);
+    this._validateIndexName(indexName);
+    this.props.onIndexNameValidationStart();
+  };
+
+  _validateIndexName = _.debounce(async (indexName: string) => {
     const indexNameError = await validateIndexName(indexName);
+    if (!this._isMounted || indexName !== this.props.indexName) {
```

Comment:
What's the purpose of the second part of this condition?

Reply:
There is no guarantee async requests will be returned in the order they are started. The second part of this condition ignores responses to outdated requests since the indexName we are validating no longer matches the props.indexName.

Reply:
Another reason is that the input can change while the validation is in progress. A user could enter an invalid name stop typing so the debounce triggers and then start typing in a valid name. So the results should be discarded unless they are for the current indexName.